### ISSUE TITLE
Entity ist missing if only null column is selected

### ISSUE THREAD

**Issue by markusramsak** [cakephp/cakephp#17116]
### Description

```php
$rfidChip = $this->RfidChips->find()
            ->select(['id'])
            ->contain([
                'Units' => [
                    'fields' => ['id', 'operator_id'],
                    'Operators' => [
                        'fields' => ['null_column'], // null_column is a column which has a value of null in this case
                    ],
                ],
            ])
            ->where([
                $this->RfidChips->aliasField('user_registration_token') => $chipRegistrationCode,
            ])
            ->firstOrFail();

        if ($rfidChip->unit->operator === null) {
            throw new BadRequestException('operator entity missing');
        }
```

if i change
` 'fields' => ['null_column']`
to
` 'fields' => ['id', 'null_column'] // added 'id' or any other column which is not null`

then 'operator' property is correctly present in the Unit-Entity

```php
$rfidChip->unit->operator // is correctly an Operator-Entity and not null any more
```

This seems like a bug to me.

### CakePHP Version

4.4.13

### PHP Version

8.2

**Comment by markstory** [comment]
How is the ORM supposed to identify that there was data and that you intentionally selected all nulls vs. the joins resulted in missing rows and there is no data?

**Comment by markusramsak** [comment]
I don't know the solution but I think it should somehow be solved because it would be an improvement in my opinion.
Sorry for not giving for help than this.

**Comment by markstory** [comment]
Ok. I don't know how to solve it either, I don't think it is solvable.

**Comment by dereuromark** [comment]
I wonder if the ORM could auto-add the primary keys where they would be necessary and omitted from the fields list?

**Comment by markusramsak** [comment]
or maybe the ORM could throw an exception like "Only nullable columns selected, you have to select at least one non null column otherwise the entity might be missing if all nullable columns are null.".

**Comment by dereuromark** [comment]
Yes, the exception could probably be a useful addition. A stricter way to make sure not wrong data appears as accident.

**Comment by markstory** [comment]
> I wonder if the ORM could auto-add the primary keys where they would be necessary and omitted from the fields list?

I think this is likely the safest and most useful solution. Sure we could use schema reflection to detect this scenario. However, I think it is more useful to append the primaryKey to the fieldlist. This helps users get what they want with fewer errors.
